sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -503,39 +504,59 @@ export class StudentService extends RecordDataModelService<Student> {
    * Use the information available in the authentication token to update
    * the user and student data currently on DB.
    * @param studentToken student authentication token.
+   * @param options method options:
+   * - `userId` user id provided in case the student data
+   * is updated by the ministry.
    * @returns updated student, if some data was changed.
    */
-  async synchronizeFromUserToken(
-    studentToken: StudentUserToken,
+  async updateStudentUserData(
+    studentUserData: StudentUserData,
+    options?: { userId?: number },
   ): Promise<Student> {
-    const studentToSync = await this.getStudentById(studentToken.studentId);
+    const userId = studentUserData.userId
+      ? studentUserData.userId
+      : options?.userId;
+    const studentToSync = await this.getStudentById(studentUserData.studentId);
     let mustSave = false;
-    if (studentToken.givenNames === undefined) {
-      studentToken.givenNames = null;
+    if (studentUserData.givenNames === undefined) {
+      studentUserData.givenNames = null;
     }
     if (
-      !dayjs(studentToken.birthdate).isSame(studentToSync.birthDate) ||
-      studentToken.lastName !== studentToSync.user.lastName ||
-      studentToken.givenNames !== studentToSync.user.firstName
+      !dayjs(studentUserData.birthdate).isSame(studentToSync.birthDate) ||
+      studentUserData.lastName.toLowerCase() !==
+        studentToSync.user.lastName.toLowerCase() ||
+      studentUserData.givenNames.toLowerCase() !==
+        studentToSync.user.firstName.toLowerCase()
     ) {
-      studentToSync.birthDate = studentToken.birthdate;
-      studentToSync.user.lastName = studentToken.lastName;
-      studentToSync.user.firstName = studentToken.givenNames;
+      studentToSync.birthDate = studentUserData.birthdate;
+      studentToSync.user.lastName = studentUserData.lastName;
+      studentToSync.user.firstName = studentUserData.givenNames;
       const sinValidation = new SINValidation();
       sinValidation.student = studentToSync;
       sinValidation.sin = studentToSync.sinValidation.sin;
       studentToSync.sinValidation = sinValidation;
       mustSave = true;
     }
     // This condition is not added above, as email does not trigger SIN validation request.
-    if (studentToken.email !== studentToSync.user.email) {
-      studentToSync.user.email = studentToken.email;
+    if (studentUserData.email !== studentToSync.user.email) {
+      studentToSync.user.email = studentUserData.email;
       mustSave = true;
     }
 
     if (mustSave) {
-      studentToSync.modifier = { id: studentToken.userId } as User;
-      return await this.save(studentToSync);
+      return this.dataSource.transaction(async (transactionalEntityManager) => {
+        await this.noteSharedService.createStudentNote(
+          studentToSync.id,
+          NoteType.General,
+          "Student information updated.",
```

Comment:
The business AC states the Ministry user would be adding a note.
```
Anytime a single, or multiple fields are edited, Ministry user must add a note. (General)
```
Also, the technical section also states a note entry. Is it missing?
![image](https://github.com/user-attachments/assets/bbd62e3c-4982-44cd-85bb-1a8be343f36b)

Reply:
Added the note